app/src/main/java/com/sdp/movemeet/backend/firebase/storage/StorageImageManager.java

```diff
@@ -0,0 +1,49 @@
+package com.sdp.movemeet.backend.firebase.storage;
+
+import android.net.Uri;
+
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.StorageTask;
+import com.sdp.movemeet.backend.BackendManager;
+import com.sdp.movemeet.backend.providers.BackendInstanceProvider;
+import com.sdp.movemeet.models.Image;
+
+public class StorageImageManager implements BackendManager<Image> {
+
+    private final StorageReference storageReference;
+
+    public StorageImageManager() {
+        this.storageReference = BackendInstanceProvider.getStorageInstance().getReference();
+    }
+
+    @Override
+    public StorageTask add(Image image, String path) {
+        return storageReference.child(path).putFile(image.getImageUri());
+    }
+
+    @Override
+    public Task<?> set(Image object, String path) {
+        return null;
```

Comment:
If this operation is unsupported, maybe you can throw a `UnsupportedOperationException`?

Reply:
Good point! I just modified it! Thanks Kepler, that's really thanks to your remark that I learn the right and proper ways to code in Java!